### ISSUE TITLE
theme pull causing whitespace changes between mac os and windows

### ISSUE THREAD

**Issue by lesterdefreitas** [Shopify/shopify-cli#1835]
# Issue summary

We have found that Shopify CLI seems to create whitespace changes when collaborators are working on mac os and windows.

## Expected behavior

Shopify CLI should ignore whitespace changes (the differences in line endings between mac os and windows) so that git does not light up all the files as having changed.

## Actual behavior

What actually happens?

A typical scenario involves one collaborator working on a mac running `shopify theme push` to deploy code to a shopify theme. Then, later, another collaborator working on windows runs `shopify theme pull` to fetch the latest code and update the repository's `main` branch before he merges work into it for deployment. At that point all the files pulled by Shopify CLI are lit up in github with whitespace changes, making it almost impossible for him to successfully merge his changes into the branch.

NOTE: This issue only affects themes that do not support the github integration and therefore require us to manually `shopify theme pull` to update our code and manually `shopify theme push` to deploy it. It also does not seem to happen if we use themekit to pull and deploy the code to and from themes across operating systems. So this is what leads us to believe that the issue is actually with Shopify CLI and how it handles files.

## Steps to reproduce the problem

1. A user using mac os runs `shopify theme deploy` to deploy code from a git branch to a theme on shopify
2. Another user on windows runs `shopify theme pull` while in that same branch to download the files from that theme into the branch

## Specifications

- Operating System: Windows, Mac OS
- Shell: Command-prompt, Terminal

**Comment by lesterdefreitas** [comment]
**UPDATE:** A possible solution to this is to add a `.gitattributes` file to the repository that normalises line endings between os's. The file's contents could look something like this:

```
# .gitattributes snippet to force users to use same line endings for project.
#
# Handle line endings automatically for files detected as text
# and leave all files detected as binary untouched.
* text=auto

#
# The above will handle all files NOT found below
# https://help.github.com/articles/dealing-with-line-endings/
# https://github.com/Danimoth/gitattributes/blob/master/Web.gitattributes

# These files are text and should be normalized (Convert crlf => lf)
*.php text
*.css text
*.js text
*.json text
*.htm text
*.html text
*.xml text
*.txt text
*.ini text
*.inc text
*.pl text
*.rb text
*.py text
*.scm text
*.sql text
.htaccess text
*.sh text
*.svg text
*.liquid text

# These files are binary and should be left untouched
# (binary is a macro for -text -diff)
*.png binary
*.jpg binary
*.jpeg binary
*.gif binary
*.ico binary
*.mov binary
*.mp4 binary
*.mp3 binary
*.flv binary
*.fla binary
*.swf binary
*.gz binary
*.zip binary
*.7z binary
*.ttf binary
*.pyc binary
```

Still, it would be great if CLI did this by default like Themekit seems to have done in the past.

**Comment by binarymonkey84** [comment]
See https://github.com/Shopify/themekit/issues/708#issue-618022013 for a similar issue but with the older `theme` CLI interface, not sure if that was fixed in that CLI since, but not considered in the `shopify` CLI suite

**Comment by lesterdefreitas** [comment]
@binarymonkey84 Thanks for the input. I can confirm that in our testing, themekit does not cause any whitespace issues when pulling files between os's. We have found though that adding the .gitattributes file to any affected repo's does solve the problem, so definitely give that a go! Still, like you said, this should be implemented within the cli itself ideally! So we'll wait on that.

**Comment by karreiro** [comment]
Closing in favor of https://github.com/Shopify/shopify-cli/issues/1453 :) (also thanks for the details provided here).
